packages/connect/src/device/Device.ts

```diff
@@ -532,6 +538,16 @@ export class Device extends TypedEmitter<DeviceEvents> {
         const { message } = await this.getCommands().typedCall('GetFeatures', 'Features', {});
         this._updateFeatures(message);
 
+        if (
+            // The check was not yet performed
+            this.authenticityChecks.firmwareRevision === null ||
+            // The check was performed, but outcome cannot be surely determined (Suite is offline)
+            // -> recheck on every getFeatures() until the result is known
+            this.authenticityChecks.firmwareRevision.success === null
+        ) {
+            await this.checkFirmwareRevision();
```

Comment:
note: what happens if we run this on some very slow internet? maybe there should be time limit.

Reply:
When I added 

It got stucked on this for 60 seconds:
![image](https://github.com/user-attachments/assets/a04bb354-54c9-4a2c-ac23-1fd84ab328d2)

- Then it begins to do discovery...
- and after a second or two you get Ghosts.

Reply:
@mroz22 Finally I was able to test and simulate it.

![image](https://github.com/user-attachments/assets/81bd8b7f-7d11-478d-b459-3cf6c6de9171)
![image](https://github.com/user-attachments/assets/8e6d60c5-4fb7-4a9b-88d8-9d307ff09c4f)

With this setup:

![image](https://github.com/user-attachments/assets/a4a2dc4b-dcd9-4090-b79c-b6da1c0c77b4)

Reply:
fair enough